wave_lang/kernel/wave/codegen/handlers.py

```diff
@@ -1669,7 +1704,20 @@ def handle_cast(emitter: WaveEmitter, node: fx.Node):
     conversion_op = get_conversion_op(
         src_elem_type, dst_elem_type, fastmath=get_fast_math_flags(emitter.options)
     )
-    casted_vector = conversion_op(dst_vector_type, vector_src)
+    if emitter.options.scalarize_packed_math and _near_mma(node):
```

Comment:
Thoughts of getting rid of `_near_mma` since we already giving kernel author the option over it by letting them set `scalarize_packed_math`?

Reply:
No point to scalarize if we don't have mfma

Reply:
cool cool, it's interesting to see that we also need scalarize for cast haha